qiskit/ignis/verification/randomized_benchmarking/rb_utils.py

```diff
@@ -59,34 +64,110 @@ def count_gates(qobj, basis, qubits):
+                transpiled_circuits_list.append(rb_circs_transpiled)
+
+            # count gate per Clifford
+            ngates = rb.rb_utils.gates_per_clifford(
+                transpiled_circuits_list=transpiled_circuits_list,
+                clifford_lengths=xdata[0],
+                basis=basis, qubits=[0, 1])
+
+            pprint.pprint(ngates)
+
+        The gate counts for qubit 0 (1) is obtained by ``ngates[0]`` (``ngates[1]``)
+        as usual python dictionary. If all gate counts are zero,
+        you might specify wrong ``basis`` or input circuit list is not transpiled into basis gates.
 
     Args:
-        qobj_list: compiled qobjs for each seed.
-        clifford_length: number of cliffords in each circuit.
-        basis: gates basis for the qobj.
-        qubits: qubits to count over.
+        transpiled_circuits_list: List of transpiled RB circuit for each seed.
+        clifford_lengths: number of Cliffords in each circuit
+        basis: gates basis for the qobj
+        qubits: qubits to count over
+        qobj_list: Deprecated. see ``transpiled_circuits_list``
+        clifford_length: Deprecated. see ``clifford_lengths``
 
     Returns:
-        l x m list of number of gates per clifford
-        (same order as basis).
-
-                * l: number of qubits,
-                * m: length of basis.
+        Nested dictionary of gate counts per Clifford.
     """
+    if qobj_list is not None:
+        transpiled_circuits_list = qobj_list
+        warn('The argument `qobj_list` will be deprecated. Use `transpiled_circuit_list`.',
+             DeprecationWarning)
 
-    ncliffs = 0
-    ngates = np.zeros([len(qubits), len(basis)], dtype=int)
+    if clifford_length is not None:
+        clifford_lengths = clifford_length
+        warn('The argument `clifford_length` will be deprecated. Use `clifford_lengths`.',
+             DeprecationWarning)
 
-    for qobj_seed in qobj_list:
-        ngates_tmp = count_gates(qobj_seed, basis, qubits)
-        for cliff_idx, cliff in enumerate(clifford_length):
-            ncliffs += (cliff + 1)  # include inverse
-            ngates += np.array(ngates_tmp[cliff_idx])
+    ncliffs = 0
+    ngates = {qubit: {base: 0 for base in basis} for qubit in qubits}
+
+    for transpiled_circuits in transpiled_circuits_list:
+        if isinstance(transpiled_circuits, list):
+            for ncliff, transpiled_circuit in zip(clifford_lengths, transpiled_circuits):
+                for instr, qregs, _ in transpiled_circuit.data:
+                    for qreg in qregs:
+                        try:
+                            ngates[qreg.index][instr.name] += 1
+                        except KeyError:
+                            pass
+                # include inverse
+                ncliffs += ncliff + 1
+        else:
+            warn('`QasmQobj` input will be deprecated. Use `QuantumCircuit` instead. '
```

Comment:
This warning doesn't need to be raised, since we have already raised a deprecation warning for using Qobj.